src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -288,6 +298,46 @@ public static int getFuelValue(ItemStack itemStack)
         }
         return fuelValue;
     }
+    
+    /**
+     * Links a set of parts to an {@link IPartFactory} that can produce them.
+     */
+    public static void registerPartFactory(IPartFactory factory, ResourceLocation... names)
+    {
+        MultipartRegistry.registerPartFactory(factory, names);
+    }
+    
+    /**
+     * Links a set of parts to an {@link IAdvancedPartFactory} that can produce them.
+     */
+    public static void registerPartFactory(IAdvancedPartFactory factory, ResourceLocation... names)
```

Comment:
To create different parts based on NBT and/or packet data - see https://github.com/CharsetMC/Charset/blob/1.9/src/main/java/pl/asie/charset/wires/logic/PartWireProvider.java